python/ray/util/sgd/v2/backends/torch.py

```diff
@@ -140,3 +141,26 @@ def on_shutdown(self, worker_group: WorkerGroup,
         if len(worker_group) > 1:
             worker_group.execute(dist.destroy_process_group)
         worker_group.execute(shutdown_torch)
+
+    def convert_checkpoint(self, checkpoint: Dict) -> Any:
+        """Convert checkpoint to stream of bytes.
+
+        This prevents deserialization errors if driver is CPU only but
+        workers are on GPU.
+        """
+        _buffer = io.BytesIO()
+        torch.save(checkpoint, _buffer)
+        return _buffer.getvalue()
+
+    def unconvert_checkpoint(self, data: Any, to_gpu: bool = False) -> Dict:
+        """Converts stream of bytes back to a checkpoint Dict.
+
+        Converts to GPU if ``to_gpu`` is True and CUDA is available.
+        """
+        _buffer = io.BytesIO(data)
+        to_gpu = to_gpu and torch.cuda.is_available()
+        state_dict = torch.load(
+            _buffer,
+            map_location=("cpu" if not to_gpu else
+                          lambda storage, loc: storage.cuda()))
```

Comment:
What's the benefit of loading this in GPU memory on the driver?

Reply:
The idea here was that if the user checkpoints a GPU model, we should still keep it as GPU unless no GPUs are available on the driver machine.

Reply:
This brings up a good point though which I didn't consider- what if the checkpointed model on the worker side is on CPU? Then we should still keep it as CPU on the driver side even if there's GPU available. Currently though there's no easy way to identify this from the bytes.